core/psh/sysexec.c

```diff
@@ -25,6 +27,63 @@ void psh_sysexecinfo(void)
 }
 
 
+static int psh_sysexec_wldargmatch(char *wld, int argc, char **argv, char wldend, int argmaxlen)
+{
+	int len = 0, ii;
+	char *currc = wld, *nextc, *cmdend;
+
+	cmdend = strchr(wld, wldend);
+	currc = wld;
+	for (ii = 0; ii < argc; ii++) {
+
+		if ((nextc = strchr(currc, ' ')) == NULL || nextc > cmdend)
+			nextc = cmdend;
+		if ((nextc == cmdend) ^ (ii == (argc - 1))) /* exits if: last argc XOR last wldcrd argument */
+			return -1;
+
+		len = nextc - currc;
+		if (len > argmaxlen || len == 0 || strlen(argv[ii]) != len || memcmp(argv[ii], currc, len) != 0)
+			return -1;
+		if (*nextc != wldend && nextc != NULL)
+			currc = nextc + 1;
+	}
+	
+	return 1;
+}
+
+
+static int psh_sysexec_checkcommand(int argc, char **argv)
+{
+	int maxcdmlen = 80, acc = -1;
+	char wldbuff[maxcdmlen], *wldlist = PSH_WILDCARDS, *curr = wldlist, *nxtcomma;
+	FILE *wldfile;
+
+	/* check against /etc/wldcrds file */
+	if ((wldfile = fopen("/etc/wldcrds", "r")) != NULL) {
```

Comment:
I would not want to use this functionality on full-featured system (with rootfs), I vote for removing reading commands from file. 
Also: reading file with each sysexec seems wasteful (I assume it should never change during runtime - if it changes we have security issue :)).

Reply:
You are right for we do not have user priviledges. But with a cenario:
_Updating software on energy meter so that you can read daytime OR nighttime energy consumption with two separate sysexecs instead of current one for all energy_

I know it can be developed in numerous ways but allowing a next command if run by privileged user may be a useful think. The other thing is when we are going to support user privileges :)

Reply:
I assume You're talking about the need to have this functionality on full-featured system - what You're talking about is business logic and it should either be taken care of in metersrv (the most secure approach), or - less secure but still better than CMD parsing - in CLI client (decide which requests can be done by which user groups). 

On full featured systems I would have access to busybox/other programs which would make possible editing/deleting file in /etc/ in multiple ways, so - as previously stated - this approach is applicable only on "small" systems. Thus, I don't see "reading from file" scenario ever happen.

If You would like to keep the "reading from file" approach, than at least the file name needs to be changed (no wildcards in there). I would also go for more readable file name :).